ykshim_client/src/test_api.rs

```diff
@@ -136,3 +139,12 @@ pub fn interpret_body<I>(body_name: &str, icx: &mut I) {
 pub fn reg_pool_size() -> usize {
     unsafe { __ykshimtest_reg_pool_size() }
 }
+
+pub fn compile_tir_trace<T>(mut tir_trace: TirTrace) -> Result<CompiledTrace<T>, CString> {
+    let compiled = unsafe { __ykshimtest_compile_tir_trace(tir_trace.0) };
+    tir_trace.0 = ptr::null_mut(); // consumed.
```

Comment:
Should we expand on this comment a bit. Something like: Since `tir_trace.0` is consumed in `__ykshimtest_compile_tir_trace`, we can set it's pointer to `null`.

Reply:
This is an idiom that appears a few times in ykshim. I considered adding a comment to the crate, but it's just standard memory management at the end of the day. Thoughts?

Reply:
Fine. I'm not too bothered by this. Just thought it might avoid some confusion in the future if the comment says a bit more than consumed. That being said, I guess I immediately knew what was going on despite not having seen this code before. So maybe it's fine as it is.

Reply:
If you are happy with this, then please kick bors.